rtCommon/bidsInterface.py

```diff
@@ -1,43 +1,292 @@
+        return streamId
+
+    def getIncremental(self, streamId, volIdx=-1) -> BidsIncremental:
+        """
+        Get a BIDS Incremental from a stream
+        Args:
+            streamId: The stream handle returned by the initXXStream call
+            volIdx: The brain volume index of the image to return. If -1 is
+                entered it will return the next volume
+        Returns:
+            A BidsIncremental containing the image volume
+        """
+        stream = self.streamMap[streamId]
+        bidsIncremental = stream.getIncremental(volIdx)
+        return bidsIncremental
+
+    def getNumVolumes(self, streamId) -> int:
+        """
+        Return the number of image volumes contained in the stream. This is only
+        defined for Bids/OpenNeuro streams (not for DicomBidsStreams)
+        Args:
+            streamId: The stream handle returned by the initXXStream call
+        Returns:
+            An int of the number of volumes
+        """
+        stream = self.streamMap[streamId]
+        return stream.getNumVolumes()
+
+
+class DicomToBidsStream():
+    """
+    A class that watches for DICOM file creation in a specified directory and with
+    a specified file pattern. When DICOM files arrive it converts them to BIDS
+    incrementals and returns the BIDS incremental. This lets a real-time classification
+    script process data directly as BIDS as it arrives from the scanner.
+    """
+    def __init__(self, dicomDir, dicomFilePattern, dicomMinSize, **entities):
+        """
+        Intialize a new DicomToBids stream, watches for Dicoms and streams as BIDS
+        Args:
+            dicomDir: The directory where the scanner will write new DICOM files
+            dicomFilePattern: A regex style pattern of the DICOM filenames to
+                watch for. They should include a {TR} tag with optional formatting.
+                For example filenames like '001_000013_000005.dcm' would have a
+                pattern '001_000013_{TR:06d}.dcm' where the volume number (TR)
+                will be filled in by a 6 digit leading zeros value.
+            dicomMinSize: Minimum size of the file to return (will continue waiting
+                if below this size)
+            entities: BIDS entities (subject, session, task, run, suffix, datatype) that
+                define the particular subject/run of the data to stream
+        """
+        # TODO - make sure dicomPattern has {TR} in it
+        # TODO - restrict allowed directories, check that dicomDir is in allowed dir
+        if 'suffix' not in entities.keys():
+            entities['suffix'] = 'bold'
+        if 'datatype' not in entities.keys():
+            entities['datatype'] = 'func'
+        self.entities = entities
+        self.dicomDir = dicomDir
+        self.dicomFilePattern = dicomFilePattern
+        self.dataInterface = DataInterface(dataRemote=False, allowedDirs=['*'], allowedFileTypes=['.dcm'])
```

Comment:
Yes, it should be more restrictive. I have a TODO for that, I wasn't sure yet the best way to pass the restricted set down, since it only applies to the DICOM case.